File: sdks/php-agent/src/Clients/LocationsClient.php
<?php
declare(strict_types=1);

namespace HMS\CarHire\Clients;

use HMS\CarHire\Config;
use HMS\CarHire\Transport\TransportInterface;

final class LocationsClient
{
    public function __construct(private TransportInterface $t, private Config $c) {}

    public function isSupported(string $agreementRef, string $locode): bool
    {
        return $this->t->isLocationSupported($agreementRef, $locode);
    }
}

File: sdks/go-agent/locations_client.go
package sdk

import (
	"context"
)

// LocationsClient provides location-related functionality
type LocationsClient struct {
	transport Transport
	config    *Config
}

// NewLocationsClient creates a new LocationsClient
func NewLocationsClient(transport Transport, config *Config) *LocationsClient {
	return &LocationsClient{
		transport: transport,
		config:    config,
	}
}

// IsSupported checks if a location is supported for a given agreement
func (lc *LocationsClient) IsSupported(ctx context.Context, agreementRef, locode string) (bool, error) {
	return lc.transport.IsLocationSupported(ctx, agreementRef, locode)
}

